mayastor/src/bdev/dev.rs

```diff
@@ -17,30 +17,39 @@
 //! Creating a bdev for any supported device type is now as simple as:
 //! ```ignore
 //!     let uri = "aio:///tmp/disk1.img?blk_size=512";
-//!     bdev::Uri::parse(&uri)?.create().await?;
+//!     bdev::uri::parse(&uri)?.create().await?;
 //! ```
 
-use std::{collections::HashMap, convert::TryFrom};
-
-use snafu::ResultExt;
-use url::Url;
+use std::collections::HashMap;
 
+use super::nvmx;
 use crate::{
-    bdev::{BdevCreateDestroy, SpdkBlockDevice, Uri},
+    bdev::SpdkBlockDevice,
     core::{BlockDevice, BlockDeviceDescriptor, CoreError},
-    nexus_uri::{self, NexusBdevError},
+    nexus_uri::NexusBdevError,
 };
 
-use super::{aio, loopback, malloc, null, nvme, nvmx, uring};
+use url::Url;
+
+pub(crate) mod uri {
+    use std::convert::TryFrom;
+
+    use snafu::ResultExt;
+
+    use crate::{
+        bdev::BdevCreateDestroy,
+        nexus_uri::{self, NexusBdevError},
+    };
+
+    use super::super::{aio, loopback, malloc, null, nvme, nvmx, uring};
```

Comment:
might tempted to suggest `crate::bdev::` instead of `super::super::`, not sure what's better